datapower-docker-compose-dp-mq/datapower-docker-compose-dp-mq.mkd

```diff
@@ -0,0 +1,168 @@
+This sample demonstrates how to utilize multiple containers, orchestrated by Docker Compose, to create an environment that includes DataPower receiving and processing a request and writing the resulting message to an MQ back end. The composed application includes everything you need -- the load driver, the DataPower, and the MQ back-end. Everything is there from config, to stylesheets, to docker-compose.yml. In this sample, I'll outline the travels of the request message, `bob`, through the docker-composed application.
+
+This is a highly customizable example and it is encouraged that you play around with it.
+
+## Prerequisites
+Before you attempt this tutorial, please be sure that you:
+
+- Read, and better yet attempt, the [Docker Compose Hello World Application](https://github.com/ibm-datapower/datapower-tutorials/blob/master/datapower-docker-compose-hello-world/datapower-docker-compose-hello-world.md) sample. This is a great starting point that offers some of the detailed framework of how DataPower containers can be created and used in a continuous integration system.
+- For your convenience, I created and used a Docker MQ base image. If you wish to run this step manually, please follow the instrucitons on [ibmcom/mq](https://hub.docker.com/r/ibmcom/mq/) on Docker Hub to create your own base Docker MQ image.
+
+
+## The Composed Application
+No rock band here, just the message Bob. Harley's Hello World example (mentioned in the Prerequisite section) goes into great detail about the various elements of his project. In order to avoid redundancy, I will not go into such detail here, but I will still cover the whole project. Let's take a look at the `docker-compose.yml` file that orchestrates the entire environment.
+```
+version: '2'
+services:
+    mq:
+      build: mq
+    http:
+      build: http
+    datapower:
+      build: datapower
+      depends_on:
+       - mq
+       - http
+    soapui:
+      image: ddavison/soapui:latest
+      depends_on:
+       - datapower
+    curldriver:
+      build: curldriver
+      depends_on:
+       - soapui
+```
+
+In all, the application consists of five services, which build five images, and run five containers. 
+
+- The `curldriver` service connects to `soapui` which drives the transaction. All curldriver does is send simple curl requests.
+- The `soapui` service uses an off-the-shelf SoapUI image from Docker Hub and adds configuration required to test the DataPower MQ service. The SoapUI request is triggered by `curldriver`, and it connects to DataPower.
+- The `datapower` service uses a DataPower Gateway from ibmcom/datapower. Configurations are added, including an application domain that contains a multi-protocol gateway. The service processes requests from `soapui`, using stylesheets from `http`, and sends the request to `mq`
+- The `http` service uses an apache server image from Docker Hub. Stylesheets and schema are placed on the server, which are called by `datapower`.
```

Comment:
Why would you bother doing that when you could just put them on local and call it a day? This is a moving part, why is it there?